legion/engine/physics/physx/systems/physx_physics_system.cpp

```diff
@@ -5,56 +5,96 @@ namespace legion::physics
 {
     using namespace physx;
 
-    PxDefaultAllocator		gAllocator;
-    PxDefaultErrorCallback	gErrorCallback;
+    PhysXPhysicsSystem::physXStatics PhysXPhysicsSystem::sPhysicsStatics;
+    size_type PhysXPhysicsSystem::sSelfInstanceCounter = 0;
 
-    PxFoundation* gFoundation = nullptr;
-    PxPvd* gPvd = NULL;
-
-    PxPhysics* gPhysics = NULL;
-
-    PxDefaultCpuDispatcher* gDispatcher = NULL;
+    PxPhysics* gPhysics = nullptr;
 
     //PhysX PVD Debugger Related
     constexpr char* pvdHost = "127.0.0.1";
     constexpr size_type defaultPVDListeningPort = 5425;
     constexpr size_type defaultPVDHostTimeout = 10;
 
+    PhysXPhysicsSystem::PhysXPhysicsSystem()
+    {
+        sSelfInstanceCounter++;
+    }
+
     void PhysXPhysicsSystem::setup()
     {
-        setupPhysXGlobalVariables();
+        lazyInitPhysXVariables();
         setupDefaultScene();
-        
+
         createProcess<&PhysXPhysicsSystem::fixedUpdate>("Physics", m_timeStep);
     }
 
-    void PhysXPhysicsSystem::fixedUpdate(time::time_span<fast_time> deltaTime)
+    PhysXPhysicsSystem::~PhysXPhysicsSystem()
```

Comment:
this should prolly be the `shutdown` function instead, since systems might be restarted without being deconstructed